File: javascript/examples/get_filing_methods.js
const baseRequest = require('../base_request.js').baseRequest

const path = require('path');
process.chdir(path.resolve(__dirname, '../../'));
require('dotenv').config();

const COMPANY_ID	= process.env.COMPANY_ID;
const JURISDICTION 	= process.env.JURISDICTION;
const PRODUCT_ID 	= process.env.FILING_PRODUCT_ID;

params = {
	company_id: COMPANY_ID,
  	jurisdiction: JURISDICTION,
  	filing_product_id: PRODUCT_ID
}

token = baseRequest.request.token({ path: '/filing-methods' });

baseRequest.request.get({ path: '/filing-methods', token: token, queryParams: params });

File: python/examples/get_filing_methods.py
import pprint
import os
import sys

sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))

from request import BaseRequest
from dotenv import dotenv_values

config = dotenv_values()

# Example of GET /filing-methods

class GetFilingMethodsRequest(BaseRequest):
    def __init__(self):
        super().__init__()
    
    def get_filing_methods(self, company_id, product_id, jurisdiction):
        params = {
            'company_id': company_id,
            'filing_product_id': product_id,
            'jurisdiction': jurisdiction
        }
        return self.make_request('GET', '/filing-methods', params=params)

company_id = config['COMPANY_ID']
product_id = config['FILING_PRODUCT_ID']
jurisdiction = config['JURISDICTION']

request = GetFilingMethodsRequest()
response = request.get_filing_methods(company_id, product_id, jurisdiction)

pprint.pprint(response)